frontend/public/components/storage/attach-storage.tsx

```diff
@@ -0,0 +1,468 @@
+      selectedKey={selectedKey}
+      required={required}
+    />
+  );
+};
+
+// Helper functions for storage operations
+const createVolume = (name, claimName) => ({
+  name,
+  persistentVolumeClaim: {
+    claimName,
+  },
+});
+
+const createVolumeMount = (name, mountPath, subPath, readOnly) => {
+  const mount: any = {
+    name,
+    mountPath,
+    readOnly,
+  };
+
+  if (subPath) {
+    mount.subPath = subPath;
+  }
+
+  return mount;
+};
+
+// Gets the mount paths currently defined in the pod template.  An
+// optional filter function for matching specific containers can be used.
+const getMountPaths = (podTemplate, /* optional function */ matchContainer) => {
+  const mountPaths = [];
+  const containers = _.get(podTemplate, 'spec.containers', []);
+  _.each(containers, function(container) {
+    if (matchContainer && !matchContainer(container)) {
+      return;
+    }
+
+    const mounts = _.get(container, 'volumeMounts', []);
+    _.each(mounts, function(mount) {
+      mountPaths.push(mount.mountPath);
+    });
+  });
+
+  return mountPaths;
+};
+
+//Return kind type give the plural
+const getKindFromPlural = plural => {
+  const ko = _.find(k8sModels, model => model.plural === plural);
+  return ko.kind;
+};
+
+const AttachStorageForm = () =>
+  class StorageFormComponent extends React.Component<
+    StorageProps,
+    StorageState
+    > {
+    constructor(props) {
+      super(props);
+      const existingResource = _.pick(props.obj, ['metadata', 'spec']);
+      const resourceObj = _.defaultsDeep({}, props.fixed, existingResource, {
+        apiVersion: 'v1',
+        data: {},
+        kind: props.kind,
+        metadata: {
+          name: '',
+        },
+        spec: {},
+      });
+
+      this.state = {
+        resourceObj,
+        inProgress: false,
+        formData: {},
+        readOnly: false,
+        disableForm: false,
+        checkboxItems: [
+          {
+            value: 'Mount as read-only',
+          },
+        ],
+        selectedOptions: [],
+      };
+    }
+
+    componentDidMount() {
+      const { kind, metadata } = this.state.resourceObj;
+      const ko = _.find(k8sModels, model => model.plural === kind);
+      const supportedKinds = [
+        'Deployment',
+        'DeploymentConfig',
+        'ReplicaSet',
+        'ReplicationController',
+      ];
+
+      if (!_.includes(supportedKinds, ko.kind)) {
+        this.setState({
+          error: `Storage is not supported for kind ${ko.kind}.`,
+          inProgress: false,
+        });
+        return;
+      }
+
+      // Get the current resource so we can add to its definition
+      k8sGet(ko, metadata.name, metadata.namespace).then(
+        obj => {
+          this.setState({ resourceObj: obj });
+        },
+        err => this.setState({ error: err.message })
+      );
+    }
+
+    onError = error => {
+      this.setState({
+        error,
+        inProgress: false,
+      });
+    };
+
+    handleChange = event => {
```

Comment:
```ts
handleChange: React.ReactEventHandler<HTMLInputElement> = event => {
```

We should add this to other event handlers as well.

Reply:
This is not working for me.  The static analyzer says there is no event.target, so I used event.currentTarget.  But that throws an unknown type error during runtime.

Reply:
> This is not working for me. The static analyzer says there is no event.target, so I used event.currentTarget. But that throws an unknown type error during runtime.

You probably want `event.currentTarget` in most cases

Is it always used for an input element? If it's used for dropdowns or other types, you'd need a different signature.

Reply:
It works for my inputs.  My dropdown is not working with that.